src/style/colors.js

```diff
@@ -81,22 +77,52 @@ class Colors {
         return color;
       }
       return requiredColor;
-    } else { // Handles dynamic colors (non uilib colors)
-      let tintLevel = Math.floor(Number(tintKey) / 10);
-      tintLevel = Math.max(1, tintLevel);
-      tintLevel = Math.min(8, tintLevel);
-      
-      if (tintLevel === BASE_COLOR_LEVEL) {
-        return color;
-      } else if (tintLevel <= BASE_COLOR_LEVEL) {
-        const darkRatio = darkRatios[tintLevel - 1];
-        return Color(color).darken(darkRatio).hex();
-      } else {
-        const lightRatio = lightRatios[tintLevel - 4];
-        return Color(color).mix(Color('#ffffff'), lightRatio).hex();
-      }
+    } else { 
+      // Handles dynamic colors (non uilib colors)
+      let tintLevel = Math.floor((Number(tintKey) / 10) + 1);
+      tintLevel = Math.max(2, tintLevel);
+      tintLevel = Math.min(9, tintLevel);
+      return generateColorTint(color, tintLevel * 10);
     }
   }
+
+  generateColorPalette(color) {
+    const hsl = Color(color).hsl();
+    const lightness = Math.round(hsl.color[2]);
+    
+    const ls = [lightness];
+    let l = lightness;
+    do {
+      l -= 10;
+      if (l > 20) {
+        ls.push(l);
+      }
+    } while (l > 20);
+
+    l = lightness;
+    do {
+      l += 10;
+      if (l < 100) {
+        ls.push(l);
+      }
+    } while (l < 100);
+
+    ls.sort((a, b) => a - b);
```

Comment:
You can use `ls.unshift(l)` instead of using `push` and this way avoid sorting the list afterward. 
`unshift` will add items to the array from the beginning.

Reply:
I will still have to sort since I don't know the position of my given value and I add it in the beginning.

Reply:
but the given position is the middle (not really middle, but between the low and the high values) of the array and you add numbers before it and numbers after it, this way it should already be sorted, no?

Reply:
oh, you mean to use it only for the first do-while? That can work...

Reply:
Done.